sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CombineFnTester.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.testing;
+
+import static org.junit.Assert.assertThat;
+
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.beam.sdk.transforms.Combine.CombineFn;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+/**
+ * Utilities for testing CombineFns, ensuring they give correct results
+ * across various permutations and shardings of the input.
+ */
+public class CombineFnTester {
+  public static <InputT, OutputT> void testCombine(
+      CombineFn<InputT, ?, OutputT> combineFn, OutputT expected, InputT... inputs) {
+    testCombine(combineFn, expected, Lists.newArrayList(inputs));
+  }
+
+  public static <InputT, OutputT> void testCombine(
+      CombineFn<InputT, ?, OutputT> combineFn, OutputT expected, Iterable<InputT> inputs) {
+    checkCombineFn(combineFn, Lists.newArrayList(inputs), expected);
+  }
+
+  // //////////////////////////////////////////////////////////////////////////
+  private static <InputT, AccumT, OutputT> void checkCombineFn(
+      CombineFn<InputT, AccumT, OutputT> fn, List<InputT> input, final OutputT expected) {
+    checkCombineFn(fn, input, Matchers.is(expected));
```

Comment:
In the future, I'd prefer `equalTo` because it shares more characters with "equals" while the natural interpretation of "is" to me is object identity. A coder "is" another coder only if they really are the same object.